prow/metrics/prowjobs/collector.go

```diff
@@ -0,0 +1,172 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package prowjobs
+
+import (
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/client-go/tools/cache"
+
+	prowapi "k8s.io/test-infra/prow/apis/prowjobs/v1"
+)
+
+type prowJobLifecycleCollector struct {
+	HistogramVec *prometheus.HistogramVec
+}
+
+func (pjc *prowJobLifecycleCollector) Update(oldJob *prowapi.ProwJob, newJob *prowapi.ProwJob) {
+	if oldJob == nil {
+		return
+	}
+	if oldJob.Status.State != newJob.Status.State {
```

Comment:
Thats definitely a very nitpicky comment, but generally its nice to keep the happy path left-aligned and avoid arrow code, i.E. instead of putting everything inside the `oldJob.Status.State != newJob.Status.State `, do a if `oldJob.Status.State == newJob.Status.State { return } `

Reply:
Done.